user-service/src/main/java/com/devspark/userservice/service/impl/UserInfoServiceImpl.java

```diff
@@ -58,4 +62,25 @@ public GetRecomUserListVO getMyRecommendUserList(GetRecomUserListDTO getRecomUse
     public List<UserInfoEntity> getAllUserInfo() {
         return userInfoRepository.findAll();
     }
+
+    @Override
+    public List<UserInfoEntity> getUserList(UserListQueryDTO userListQueryDTO) {
+        // Default not pageable
+        Pageable pageable = Pageable.unpaged();
+        // If page and pageSize are not null, then pageable
+        // Example: page = 1, pageSize = 10
+        if (userListQueryDTO.getPage() != null && userListQueryDTO.getPageSize() != null) {
+            // If orderBy and direction are not null, then sort
+            // Example: orderBy = "id", direction = "DESC"
+            if (userListQueryDTO.getOrderBy() != null && userListQueryDTO.getDirection() != null
+            && !userListQueryDTO.getOrderBy().isEmpty() && !userListQueryDTO.getDirection().isEmpty()){
+                Sort sort = Sort.by(userListQueryDTO.getDirection().equals("ASC") ?
+                        Sort.Direction.ASC : Sort.Direction.DESC,
+                        userListQueryDTO.getOrderBy());
+                pageable = PageRequest.of(userListQueryDTO.getPage() - 1, userListQueryDTO.getPageSize(), sort);
+            }else
+                pageable = PageRequest.of(userListQueryDTO.getPage() - 1, userListQueryDTO.getPageSize());
+        }
+        return userInfoRepository.getUserList(userListQueryDTO.getUserIds(), pageable);
```

Comment:
For pagination I highly recommend using native query since it will be easier to read and maintenance.